routers/api/v1/api.go

```diff
@@ -1399,8 +1399,8 @@ func Routes() *web.Router {
 				}, mustAllowPulls, reqRepoReader(unit.TypeCode), context.ReferencesGitRepo())
 				m.Group("/statuses", func() {
 					m.Combo("/{sha}").Get(repo.GetCommitStatuses).
-						Post(reqToken(), reqRepoWriter(unit.TypeCode), bind(api.CreateStatusOption{}), repo.NewCommitStatus)
-				}, reqRepoReader(unit.TypeCode))
+						Post(reqToken(), tokenRequiresScopes(auth_model.AccessTokenScopeCategoryCommitStatus, auth_model.AccessTokenScopeCategoryRepository), bind(api.CreateStatusOption{}), repo.NewCommitStatus)
```

Comment:
This still does not look right as Get method (Combo) should test read rights and Post - write rights

Reply:
This should be fixed. I re-introduced `reqRepoWriter` and `Reader` middlewares. I'm getting an error that repository permissions are required. So there is something I don't understand in how these middlewares work.

Is there a way to write tests for routing?